service/db/src/main/java/gov/va/vro/service/db/SaveToDbServiceImpl.java

```diff
@@ -70,7 +73,9 @@ public void insertAssessmentResult(
   @Override
   public void insertEvidenceSummaryDocument(GeneratePdfPayload request, String documentName) {
     ClaimEntity claim =
-        claimRepository.findByClaimSubmissionId(request.getClaimSubmissionId()).orElse(null);
+        claimRepository
+            .findByClaimSubmissionIdAndIdType(request.getClaimSubmissionId(), request.getIdType())
+            .orElse(null);
```

Comment:
I don't see why convert to null here. You can just check if the optional is present